### PR TITLE
Support custom lamports amount in init for pda

### LINKED ISSUE #1224: lang: generate `init` function ignores `rent-exempt = skip`, always makes account rent exempt

### PR DESCRIPTION
**Pull request by hami9x**
Important feature I think. This also fixes #1224.

### CODE REVIEW

File: lang/syn/src/codegen/accounts/constraints.rs [hunk 1]
```diff
@@ -630,31 +656,52 @@ pub fn generate_init(
...
 // Generated code to create an account with with system program with the
 // given `space` amount of data, owned by `owner`.
 //
 // `seeds_with_nonce` should be given for creating PDAs. Otherwise it's an
 // empty stream.
-pub fn generate_create_account(
+pub fn generate_create_account_lamports(
     field: &Ident,
     space: proc_macro2::TokenStream,
+    lamports: proc_macro2::TokenStream,
```
Comment (armaniferrante) [inline lang/syn/src/codegen/accounts/constraints.rs:682]: This should probably be an Option considering the use of `proc_macro2::TokenStream::new()` above.

File: lang/syn/src/parser/accounts/constraints.rs [hunk 2]
```diff
@@ -264,7 +270,7 @@ pub fn parse_token(stream: ParseStream) -> ParseResult<ConstraintToken> {
                         error: parse_optional_custom_error(&stream)?,
                     },
                 )),
-                _ => return Err(ParseError::new(ident.span(), "Invalid attribute")),
+                _ => return Err(ParseError::new(ident.span(), "Invalid attribute 4")),
```
Comment (armaniferrante) [inline lang/syn/src/parser/accounts/constraints.rs:273]:
```suggestion
                _ => return Err(ParseError::new(ident.span(), "Invalid attribute")),
```

File: tests/misc/programs/misc/src/context.rs [hunk 3]
```diff
@@ -86,6 +86,22 @@ pub struct TestPdaInit<'info> {
...
 
+#[derive(Accounts)]
+#[instruction(domain: String, seed: Vec<u8>, bump: u8, lamports: u64)]
+pub struct TestPdaInitCustomLamports<'info> {
+    #[account(
+        init,
+        seeds = [b"my-seed", domain.as_bytes(), foo.key.as_ref(), &seed],
+        bump = bump,
+        payer = my_payer,
+        lamports = lamports,
```
Comment (armaniferrante) [inline tests/misc/programs/misc/src/context.rs:97]: We should add this to the `#[derive(Accounts)]` api reference docs.

File: lang/syn/src/parser/accounts/constraints.rs (hunk 2 above)
Comment (chilons) [reply lang/syn/src/parser/accounts/constraints.rs:273]: My bad, I used these for testing and forgot to change it back.

**Comment by armaniferrante** [comment]
Closing as stale.

**Pull request closed without merging**